### PR TITLE
Document captured HTTP headers configuration options

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
I moved docs for properties that are common to all instrumentations to a separate file - and created a new directory for instrumentation configuration docs. Perhaps in the future we'll have a `config/reactor-netty.md` file with Reactor Netty speficic configs, a `config/spring-batch.md` with Spring Batch configs and so on.

CC @theletterf

### CODE REVIEW

File: docs/config/common.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+| System property                                       | Environment variable                                  | Description                                                         |
+|-------------------------------------------------------|-------------------------------------------------------|---------------------------------------------------------------------|
+| `otel.instrumentation.common.db-statement-sanitizer.enabled` | `OTEL_INSTRUMENTATION_COMMON_DB_STATEMENT_SANITIZER_ENABLED` | Enables the DB statement sanitization. The default value is `true`. |
+
+## Capturing HTTP request and response headers
+
+The agent can be configured to capture chosen HTTP headers as span attributes, according to the
+[semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers)
+. The following properties may be used to define which headers should be captured:
```
Comment (theletterf) [inline docs/config/common.md:40]:
```suggestion
. Use the following properties to define which HTTP headers you want to capture:
```

File: docs/config/common.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+| `otel.instrumentation.common.experimental.capture-http-headers.client.response` | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_CLIENT_RESPONSE` | A comma-separated list of HTTP header names. HTTP client instrumentations will capture HTTP response header values for all configured header names.
+| `otel.instrumentation.common.experimental.capture-http-headers.server.request`  | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_SERVER_REQUEST`  | A comma-separated list of HTTP header names. HTTP server instrumentations will capture HTTP request header values for all configured header names.
+| `otel.instrumentation.common.experimental.capture-http-headers.server.response` | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_SERVER_RESPONSE` | A comma-separated list of HTTP header names. HTTP server instrumentations will capture HTTP response header values for all configured header names.
+
+These configuration options are supported by all javaagent HTTP client and server instrumentations.
```
Comment (theletterf) [inline docs/config/common.md:49]:
```suggestion
These configuration options are supported by all HTTP client and server instrumentations.
```

File: docs/config/common.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+NOTE: the property/env variable names listed in the table above are still experimental, and thus are
```
Comment (theletterf) [inline docs/config/common.md:51]:
```suggestion
> **Note**: The property/environment variable names listed in the table are still experimental, and thus are
```

File: docs/config/common.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+subject to change!
```
Comment (theletterf) [inline docs/config/common.md:52]:
```suggestion
subject to change.
```

**Review by theletterf — CHANGES_REQUESTED** [review]

File: docs/config/common.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+| System property                                              | Environment variable                                         | Description |
+| ------------------------------------------------------------ | ------------------------------------------------------------ | ----------- |
+| `otel.instrumentation.common.db-statement-sanitizer.enabled` | `OTEL_INSTRUMENTATION_COMMON_DB_STATEMENT_SANITIZER_ENABLED` | Enables the DB statement sanitization. The default value is `true`.
+
+## Capturing HTTP request and response headers
+
+You can configure the agent to capture predefined HTTP headers as span attributes, according to the
+[semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers)
+. Use the following properties to define which HTTP headers you want to capture:
```
Comment (trask) [inline docs/config/common.md:39]:
```suggestion
[semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers).
Use the following properties to define which HTTP headers you want to capture:
```

File: docs/config/common.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+| `otel.instrumentation.common.experimental.capture-http-headers.client.response` | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_CLIENT_RESPONSE` | A comma-separated list of HTTP header names. HTTP client instrumentations will capture HTTP response header values for all configured header names.
+| `otel.instrumentation.common.experimental.capture-http-headers.server.request`  | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_SERVER_REQUEST`  | A comma-separated list of HTTP header names. HTTP server instrumentations will capture HTTP request header values for all configured header names.
+| `otel.instrumentation.common.experimental.capture-http-headers.server.response` | `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CAPTURE_HTTP_HEADERS_SERVER_RESPONSE` | A comma-separated list of HTTP header names. HTTP server instrumentations will capture HTTP response header values for all configured header names.
```
[the remark this answers is not in the document]
Comment (theletterf) [reply docs/config/common.md:46]: Wait, they’re still open for discussion? I love naming vars. Let’s discuss today. :)

**Pull request merged**